### DISCUSSION TITLE
about small triangle simplification

### DISCUSSION THREAD

**Discussion by lyonscott** [zeux/meshoptimizer#672] [General]
I have applied this library to a real project and found that if I lock the edges and add normal weighting, some very small triangles will be retained within a threshold of 0.2. Is there a possibility that I can incorporate some visual optimization for the triangles in the distant LOD since they occupy a low percentage of actual rendered pixels?

**Comment by zeux** [comment]
Do you mean individual small triangles? That likely is due to a topological constraint - either an open edge, or possibly a complex non manifold topology around one of the vertices. Note that for simplification to get less stuck it's important that the input vertices are welded to the extent that's allowed by the attributes, e.g. regions with flat shaded triangles (separate normal per face-vertex) will currently not be simplified at all due to topological restrictions.

**Comment by lyonscott** [comment]
For example, in the case of a human model with an eyeball, after mesh simplification, the eyeball may collapse into two or three triangles. However, due to its small size on the screen, it becomes difficult to see. In other mesh simplification software like Simplygon, I have noticed that it is possible to eliminate or cull such triangles using a lower threshold. 
So, if I can compute the vertex color and determine that a triangle is small on the screen, can I assign a negative weight to it and discard these triangles in the algorithm? 
Or, we can utilize the half-edge structure to calculate the value of each face. If a face has only one or two neighboring faces, it may indicate that it is located in a boundary area or is isolated. Additionally, if the area of a face is small, we can choose to discard it.
